evaluate/task/task-code-repair.go

```diff
@@ -0,0 +1,157 @@
+type TaskCodeRepairArguments struct {
+	// Mistakes holds the list of compilation errors for a repository.
+	Mistakes []string
+}
+
+var _ evaltask.Task = (*TaskCodeRepair)(nil)
+
+// newCodeRepairTask returns a code repair task.
+func newCodeRepairTask(logger *log.Logger, resultPath string, model model.Model, language language.Language) (task evaltask.Task) {
+	return &TaskCodeRepair{
+		ResultPath: resultPath,
+		Language:   language,
+		Model:      model,
+		Logger:     logger,
+	}
+}
+
+// Identifier returns the code repair task identifier.
+func (t *TaskCodeRepair) Identifier() evaltask.Identifier {
+	return IdentifierCodeRepair
+}
+
+// Run performs source code repairing in a repository with compilation errors.
+func (t *TaskCodeRepair) Run(repository evaltask.Repository) (repositoryAssessment metrics.Assessments, problems []error, err error) {
+	log, logClose, err := log.WithFile(t.Logger, filepath.Join(t.ResultPath, string(t.Identifier()), model.CleanModelNameForFileSystem(t.Model.ID()), t.Language.ID(), repository.Name()+".log"))
+	if err != nil {
+		return nil, nil, err
+	}
+	defer logClose()
+
+	var innerRepositories []string
+	files, err := os.ReadDir(repository.DataPath())
+	if err != nil {
+		return nil, nil, pkgerrors.WithStack(err)
+	}
+	for _, file := range files {
+		if file.IsDir() && !strings.HasPrefix(file.Name(), ".") {
+			innerRepositories = append(innerRepositories, filepath.Join(repository.DataPath(), file.Name()))
+		}
+	}
+
+	repositoryAssessment = metrics.NewAssessments()
+	for _, innerRepository := range innerRepositories {
+		if err := repository.Reset(t.Logger); err != nil {
+			t.Logger.Panicf("ERROR: unable to reset temporary repository path: %s", err)
+		}
+
+		filePaths, err := t.Language.Files(log, innerRepository)
+		if err != nil {
+			return nil, problems, pkgerrors.WithStack(err)
+		}
+		if len(filePaths) != 2 {
+			problems = append(problems, pkgerrors.Errorf("the inner repository must only contain an implementation file and the corresponding test file, but found %#v", filePaths))
+
+			continue
+		}
+		var hasTestFile bool
+		var sourceFile string
+		for _, file := range filePaths {
+			if strings.HasSuffix(file, t.Language.DefaultTestFileSuffix()) {
+				hasTestFile = true
+
+				continue
+			}
+			if filepath.Ext(file) == t.Language.DefaultFileExtension() {
+				sourceFile = file
+			}
+		}
+		if sourceFile == "" {
+			problems = append(problems, pkgerrors.New("could not find an implementation file"))
+
+			continue
+		}
+		if !hasTestFile {
+			problems = append(problems, pkgerrors.New("could not find a test file"))
+
+			continue
+		}
+
+		mistakes, err := t.Language.Mistakes(t.Logger, innerRepository)
+		if err != nil {
+			problems = append(problems, pkgerrors.WithStack(err))
+
+			continue
+		}
+		if len(mistakes) == 0 {
+			problems = append(problems, pkgerrors.Errorf("the following repository must contain source files with compilation errors: %q", repository.DataPath()))
```

Comment:
`test case %q in repository %q must contain source files with compilation errors`

this is also an error that we should check once in the very beginning and that should be a hard error cause if something is wrong with the layout, it will stay wrong for the whole evaluation.